src/Export/ProductServiceSeparateVariants.php

```diff
@@ -0,0 +1,302 @@
+    {
+        $criteria->addFilter(
+            new ProductAvailableFilter(
+                $this->salesChannelContext->getSalesChannel()->getId(),
+                ProductVisibilityDefinition::VISIBILITY_SEARCH
+            )
+        );
+    }
+
+    protected function addProductAssociations(Criteria $criteria): void
+    {
+        Utils::addProductAssociations($criteria);
+    }
+
+
+    protected function addPriceZeroFilter(Criteria $criteria): void
+    {
+        $criteria->addFilter(
+            new RangeFilter('price', [
+                RangeFilter::GT => 0
+            ])
+        );
+    }
+
+    /**
+     * @see \Shopware\Core\Content\Product\SalesChannel\Listing\ProductListingLoader::addGrouping()
+     */
+    protected function addGrouping(Criteria $criteria): void
+    {
+        $criteria->addGroupField(new FieldGrouping('displayGroup'));
+
+        $criteria->addFilter(
+            new NotFilter(
+                NotFilter::CONNECTION_AND,
+                [new EqualsFilter('displayGroup', null)]
+            )
+        );
+    }
+
+    /**
+     * @see \Shopware\Core\Content\Product\SalesChannel\Listing\ProductListingLoader::handleAvailableStock()
+     */
+    protected function handleAvailableStock(Criteria $criteria): void
+    {
+        $salesChannelId = $this->salesChannelContext->getSalesChannel()->getId();
+        $systemConfigService = $this->container->get(SystemConfigService::class);
+
+        $hide = $systemConfigService->get('core.listing.hideCloseoutProductsWhenOutOfStock', $salesChannelId);
+        if (!$hide) {
+            return;
+        }
+
+        $criteria->addFilter(
+            new NotFilter(
+                NotFilter::CONNECTION_AND,
+                [
+                    new EqualsFilter('product.isCloseout', true),
+                    new EqualsFilter('product.available', false),
+                ]
+            )
+        );
+    }
+
+    protected function addProductIdFilters(Criteria $criteria, string $productId): void
+    {
+        $productFilter = [
+            new EqualsFilter('ean', $productId),
+            new EqualsFilter('manufacturerNumber', $productId),
+            new EqualsFilter('productNumber', $productId),
+        ];
+
+        // Only add the id filter in case the provided value is a valid uuid, to prevent Shopware
+        // from throwing an exception in case it is not.
+        if (Uuid::isValid($productId)) {
+            $productFilter[] = new EqualsFilter('id', $productId);
+        }
+
+        $criteria->addFilter(
+            new MultiFilter(
+                MultiFilter::CONNECTION_OR,
+                $productFilter
+            )
+        );
+    }
+
+    protected function getVisibleProducts(?int $limit, ?int $offset, ?string $productId): EntitySearchResult
+    {
+        $criteria = $this->getCriteriaWithProductVisibility($limit, $offset);
+
+        $criteria->addFilter(
+            new EqualsFilter('parentId', null)
+        );
```

Comment:
I would suggest something like:

```suggestion
        $criteria = $this->getBaseCriteria($limit, $offset);
        $this->adaptCriteriaBasedOnConfiguration($criteria)
```